src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public partial class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+        private readonly JsonSerializerOptions? _jsonSerializerOptions;
+
+        private static MediaTypeHeaderValue CreateMediaTypeFromString(string mediaTypeAsString)
+        {
+            MediaTypeHeaderValue mediaType = new MediaTypeHeaderValue(mediaTypeAsString);
+            Debug.Assert(mediaType.CharSet == null);
+            mediaType.CharSet = Encoding.UTF8.WebName;
+
+            return mediaType;
+        }
+
+        public Type ObjectType { get; }
+
+        public object? Value { get; }
+
+        public JsonContent(Type type, object? value, JsonSerializerOptions? options = null)
+            : this(type, value, CreateMediaTypeFromString(JsonMediaType), options) { }
+
+        public JsonContent(Type type, object? value, string mediaType, JsonSerializerOptions? options = null)
+            : this(type, value, CreateMediaTypeFromString(mediaType ?? throw new ArgumentNullException(nameof(mediaType))), options) { }
+
+        public JsonContent(Type type, object? value, MediaTypeHeaderValue mediaType, JsonSerializerOptions? options = null)
+        {
+            if (mediaType == null)
+            {
+                throw new ArgumentNullException(nameof(mediaType));
+            }
+
+            // TODO: Support other charsets once https://github.com/dotnet/runtime/issues/30260 is done.
```

Comment:
30260 is only for .NET 5.  What's the plan for this downlevel?

Reply:
Can we copy `TranscodingRead/WriteStream` classes and use that for NS2.0 and .NET5? Then we can replace it on .NET5 once 30260 is done.
cc @pranavkm